ui/src/message_popup/author/question_editing_components/question_text_test.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import {shallow} from 'enzyme';
+import {expect} from 'chai';
+import sinon from 'sinon';
+
+import QuestionText from './question_text.jsx';
+
+import TextField from 'material-ui/TextField';
+
+import {testQuestion} from '../../test_fixtures.js';
+
+function testProps(props){
+  return ({
+    originalText: testQuestion.text,
+    questionText: testQuestion.text + ' THIS HAS BEEN EDITED!',
+    onQuestionTextChange: sinon.spy(),
+    ...props
+  });
+}
+
+describe('<EditingComponent.QuestionText />', ()=>{
+  it('renders the original question text', ()=>{
+    const props = testProps();
+    const wrapper = shallow(<QuestionText {...props} />);
+    expect(wrapper.find(".originalQuestionText").text()).to.equal(props.originalText);
+    expect(wrapper.find(TextField)).to.have.length(1);
+    expect(wrapper.find(TextField).props().value).to.equal(props.questionText);
+  });
+  it('does not render original question text if not provided', ()=>{
+    const props = testProps({originalText: undefined});
+    const wrapper = shallow(<QuestionText {...props} />);
+    expect(wrapper.find(".originalQuestionText")).to.have.length(0);
+  });
+  it('returns the changed text', ()=>{
+    const props = testProps();
+    const wrapper = shallow(<QuestionText {...props} />);
+    wrapper.find(TextField).simulate('change', { target: { value: testQuestion.text } });
+    expect(props.onQuestionTextChange.callCount).to.equal(1);
+    expect(props.onQuestionTextChange.firstCall.args).to.deep.equal([{ target: { value: testQuestion.text } }]);
```

Comment:
This is a minor point I should have noticed earlier, but I think in general it'd be better to pass just `testQuestion.text` to the callback - no need to pass it the full DOM event.  So concretely the `QuestionText` component wouldn't directly pass in `props.onQuestionTextChange` to `TextField` but would pass its own method, and then grab the value from the event and pass only that on `props.onQuestionTextChange`.

This is true for the other components that `QuestionPage` uses too.